VEuPathDB/web-monorepo
# Use filter selection as vocab for variables

### Description

Closes #138 

Variable constraints currently operation on dataset-wide characteristics of a variable. For example, if a variable has > 8 values, it will not be available in many contexts of a visualization.

This PR makes it so that if the subset is filtered by a categorical variable, then the filter selection is used in place of the dataset-wide variable's vocabulary. This makes it possible to select <= 8 values, and then use the variable in visualizations.

### Notes
- All inputs are are filter sensitive wrt `{min,max}NumValues`
- Legend labels are constrained to filter selection
- Plot ticks are constrained to filter selection
- Introduced pattern using `useDeepValue` to eliminate eslint warnings re vizConfig properties in visualizations.

### Testing notes
- Check that variable inputs and legends respond to filter changes
- Check UX of when a filter is removed and a selected variable is no longer allowed
- Check standalone map
